core/store/src/opener.rs

```diff
@@ -127,7 +127,7 @@ impl From<SnapshotRemoveError> for StoreOpenerError {
 fn get_default_kind(archive: bool, temp: Temperature) -> DbKind {
     match (temp, archive) {
         (Temperature::Hot, false) => DbKind::RPC,
-        (Temperature::Hot, true) => DbKind::Archive,
+        (Temperature::Hot, true) => DbKind::Hot,
```

Comment:
I cry on the inside every time we change this. 
Can you double check that this is backwards compatible? 
Do we have canary archival nodes to make sure it doesn't break?

Reply:
@posvyatokum I think this broke the `split_storage.py` nayduck test. I tried the simple fix of changing the temperature from archive to hot but then another issue popped up: 
```
2024-05-11T13:44:20.439468Z ERROR cold_store: cold_store_copy failed error="ColdHeadBehindHotTailError {\n    cold_head_height: 0,\n    hot_tail_height: 22,\n}"
```

That makes me think this change actually broke some invariants that we had about not moving tail until the cold migration is finished. Can you have a look at it please?